src/EiffelEvents.RabbitMq.Client/RabbitMqEiffelClient.cs

```diff
@@ -107,6 +144,61 @@ public string Subscribe<T>(string serviceIdentifier, Action<T, ulong> callback)
             }
         }
 
+        private Result<T> ValidateEvent<T>(SchemaValidationOnSubscribe validateOnSubscribe, T typeObj, string content)
```

Comment:
I suggets to rename this method to TryDeseralizeEvent (like the norm of int.TryParse() ) as it validates and deserialzes if possible.

also it is good to add its XMLDocstring